---
title: "microbiomeutilities example"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```



## Vignette Info

This package is in experimental stage and should be used only for testing/trial purposes. I will keep improving this with time and feedback.  
This is mainly a wrapper tool R package. Apart for some simple scripts for data visualization, this package has a single function called `microbiome_pipeline` for carrying out preliminary QC, Alpha Diversity, Ordination and Composition analysis of OTU tables. The output is a HTML report for convenient investigating of the data.  


## Install  

```{r, eval=FALSE}

install.packages("devtools")
devtools::install_github("microsud/microbiomeutilities")

```  


```{r}

library(microbiomeutilities)
library(microbiome)
library(knitr)
library(tibble)

```


## Example data 

A subset of human intestinal biopsy and faecal data from pre-print [Shetty SA et al. 2017](https://www.biorxiv.org/content/early/2017/12/06/227272).  

Useful resources:  

For more information you can have a look at [Phyloseq](http://joey711.github.io/phyloseq/)  
Tools for microbiome analysis in R. **Microbiome package version 1.1.2.** URL: [http://microbiome.github.com/microbiome](http://microbiome.github.com/microbiome).  


```{r}

data("biogeogut")
ps0 <- biogeogut

print(ps0)

```



## Microbiome analysis pipeline  

Function microbiome_pipeline generates an HTML report with preliminary QC, Alpha Diversity, Ordination and Composition analysis of OTU tables. This function save all intermediate files incuding figures and phyloseq objects in user specified directory.

```{r, eval=FALSE}

microbiome_pipeline(
  otufile = "my.biom",
  mapping = "mymap.csv",
  taxonomy = NULL,
  treefilename = "myTree.tre",
  type = "biom",
  work_dir = "F:/path/my/input/filefolder",
  out_dir = "F:/path/to/save/my/files/folder",
  VariableA = "MC_type1",
  VariableB = "Region",
  UnConstOrd = TRUE,
  heatmap = TRUE,
  filterCount = 4,
  filterPrev = 0.01,
  col.palette = "Paired",
  filterpseq = TRUE,
  samsize = NA,
  projectname = "Mock",
  author = "Sudarshan"
  )
```

## Formatting the Phyloseq Object  

Most commonly it is observed that the taxonomy file has classification until a given taxonomic level. We will fill the empty cells with the maximum classification available along with the OTU number.  

Check the taxonomy in phyloseq object.  

```{r}

kable(head(tax_table(ps0)))

```

Some have only g__ of s__ information.  

```{r}

ps0.f <- format_phyloseq(ps0)

kable(head(tax_table(ps0.f)))

```

Now the available taxonomy is added.  
There is a second version which will change the names in both otu table and taxonomy table. This can be useful if the analysis has to be done at OTU level. Only ID are less useful.    


```{r}

ps0.f2 <- format_to_besthit(ps0)

kable(head(tax_table(ps0.f2)))

```

As can be seen, the rownames have the OTUIDs and available toxonomic name(s).  

## Summarize the percent taxa classification for phyloseq  

This can be useful to get an overview of taxonomic classifications. Only patterns such as [g__] or similar is expected. [g__<empty>] or similar not considered pease convert for eg. g__unclassified to uniform [g__].  

```{r}

percent_classified(ps0)

```

## Distribution of reads  

Useful for QC purposes.  

```{r}

p <- plot_read_distribution(ps0, groups="SampleType", plot.type= "density")
print(p)
```


## Convert phyloseq object to long data format  

Useful in the user wants to plot specific features ot imporve base plots.  

```{r}

pseq_df <- phy_to_ldf(ps0, transform.counts = NULL)

kable(head(pseq_df))

```

## Plot alpha diversities  

Utility plot function for diversity measures calcualted by microbiome package.  

```{r, fig.width=6, fig.height=4}

library(microbiome)
data("biogeogut")
ps0 <- biogeogut

p <- plot_alpha_diversities(ps0, 
                            type = "diversities", 
                            index.val = "all", 
                            plot.type = "stripchart", 
                            variableA = "SampleType", 
                            palette = "jco")

print(p)


```


## Plot ordination and core  

```{r, fig.height=8, fig.width=12}
library(phyloseq)
library(microbiome)
data("atlas1006")
pseq <- atlas1006

pseq.rel <- microbiome::transform(pseq, "compositional")

pseq.core <- subset_samples(pseq.rel, gender == "female" & DNA_extraction_method == "r" & bmi_group == "lean")

prev.thres <- seq(.05, 1, .05)
det.thres <- 10^seq(log10(1e-2), log10(.2), length = 10)

# Currently only NMDS is supported
ord.bray <- ordinate(pseq.core, "NMDS", "bray")

px <- plot_ordiplot_core(pseq.core, ordiObject = ord.bray, coreplot = TRUE, prevalences = prev.thres, detections = det.thres, color.opt = "diversity", shape = NULL)

```


## Visualize abundance of select taxa  


```{r, fig.width=8}

ps.f2 <- format_to_besthit(ps0)
psf2.rel <- microbiome::transform(ps.f2, "compositional")
otu <- abundances(psf2.rel)
meta <- meta(psf2.rel)

library(vegan)
permanova <- adonis(t(otu) ~ SampleType,
               data = meta, permutations=99, method = "bray")

# P-value
print(as.data.frame(permanova$aov.tab)["group", "Pr(>F)"])

coef <- coefficients(permanova)["SampleType1",]
top.coef <- coef[rev(order(abs(coef)))[1:6]]

top.coef.df <- as.data.frame(top.coef)

my_taxa <- c(rownames(top.coef.df))

p <- plot_select_taxa(psf2.rel, my_taxa, "SampleType", "Paired", plot.type = "boxplot")
p
```

## Plot taxa boxplot  

Plot relative abundance of top taxa specified by user. 

```{r,fig.width=6, fig.height=4}

pn <- plot_taxa_boxplot(ps0,
                        taxonomic.level = "Phylum",
                        top.otu = 5, VariableA = "SampleType",
                        title = "Relative abudance plot", color = "Set2")

print(pn)

```

## Plot taxa barplot  

```{r, fig.width=6, fig.height=4}
library(RColorBrewer)
library(dplyr)

ps.cor <- core(ps0, detection = 10, prevalence = 0.5)
p <- plot_taxa_composition(ps.cor, taxonomic.level = "Phylum", transform = "compositional", average_by = NULL)
p
```

## Heatmap using phyloseq and pheatmap   

Useful for visualisng differences in top otus between sample groups.  

```{r, fig.width=8}

heat.sample <- plot_taxa_heatmap(ps0, subset.top = 20,
    VariableA = "SampleType",
    heatcolors =brewer.pal(100, "Blues"),
    transformation = "log10")

```


```{r}

sessionInfo()

```


## Package requirements  

* Depends:
 + [R (>= 3.4.1)](https://www.r-project.org/)
 + [phyloseq](https://joey711.github.io/phyloseq/index.html)
 + [ggplot2](http://ggplot2.org/)
 + [microbiome](https://bioconductor.org/packages/devel/bioc/html/microbiome.html)

* Imports:
 + [plyr](https://cran.r-project.org/web/packages/plyr/index.html)
 + [reshape2](https://cran.rstudio.com/web/packages/reshape2/index.html)
 + [stats](https://stat.ethz.ch/R-manual/R-devel/library/stats/html/00Index.html)
 + [tidyr](http://tidyr.tidyverse.org/)
 + [utils](https://stat.ethz.ch/R-manual/R-devel/library/utils/html/00Index.html)
 + [data.table](https://cran.r-project.org/web/packages/data.table/index.html)
 + [vegan](https://cran.r-project.org/web/packages/vegan/index.html)
 + [ggpubr](https://cran.r-project.org/web/packages/ggpubr/index.html)
 + [RColorBrewer](https://cran.r-project.org/web/packages/RColorBrewer/index.html)
 + [viridis](https://cran.r-project.org/web/packages/viridis/index.html)
 + [pheatmap](https://cran.r-project.org/web/packages/pheatmap/index.html)
 + [ggrepel](https://cran.r-project.org/web/packages/ggrepel/index.html)

* Suggests:
 + [BiocGenerics](https://bioconductor.org/packages/release/bioc/html/BiocGenerics.html)
 + [knitr](https://cran.r-project.org/web/packages/knitr/index.html)
 + [rmarkdown](http://rmarkdown.rstudio.com/)
 + [ape](https://cran.r-project.org/web/packages/ape/index.html)
 + [Picante](https://academic.oup.com/bioinformatics/article/26/11/1463/203321)
